x4.md

```diff
@@ -54,325 +49,93 @@ The protocol should be a reflection of our principles.
+The purpose of the privacy layer is to hide metadata (provide darkness) that are always sent when communicating over the internet. In some cases, leaked metadata might be as dangerous as leaking the actual payload so it's important to reduce that.
 
-# Offline messages
+The Status Protocol uses [Whisper in version 6](https://github.com/ethereum/wiki/wiki/Whisper) as a privacy protection protocol. Whisper routes messages by broadcasting them to all connected peers so that identifying of the recipient is almost impossible (the sender is not equally protected, though). [Read more](https://github.com/ethereum/go-ethereum/wiki/Achieving-Darkness) about how Whisper achieves darkness.
 
-In the case of mobile clients which are often offline, there is a strong need to have an ability to download offline messages. By offline messages, we mean messages sent into the Whisper network and expired before being collected by the recipient. A message stays in the Whisper network for a duration specified as `TTL` (time-to-live) property.
+Note: there is one more cross-layer dependency in the Status Protocol between the privacy and data layers. Whisper uses topics to reduce traffic and preliminarily filter out messages. However, selecting a topic depends on the type of the message that is a topic for 1-1 chat will be different from a topic for a public chat.
 
-A Whisper client needs to register a mail server instance which will be used by [geth's Whisper service](https://github.com/ethereum/go-ethereum/blob/v1.8.23/whisper/whisperv6/whisper.go#L209-L213).
+Please refer to [Initial Transport Privacy through Whisper Specification](x7.md) for more details.
 
-`MailServer` is an interface with two methods:
-* `Archive(env *Envelope)`
-* `DeliverMail(whisperPeer *Peer, request *Envelope)`
+# P2P layer
 
-If a mail server is registered for a given Whisper client, it will save all incoming messages on a local disk (this is the simplest implementation, it can store the messages wherever it wants, also using technologies like swarm and IPFS) in the background.
+P2P layer allows various clients to securely communicate with each other through the internet in a decetralized fashion, envetually forming a peer-to-peer network. Due to our close relationship to Ethereum and the fact that Whisper is built on top of [devp2p](https://github.com/ethereum/devp2p), we selected [devp2p](https://github.com/ethereum/devp2p) as a P2P layer.
 
-Notice that each node is meant to be independent and SHOULD keep a copy of all historic messages. High Availability (HA) can be achieved by having multiple nodes in different locations. Additionally, each node is free to store messages in a way which provides storage HA as well.
+A client in order to rely a message needs to first find other clients within the peer-to-peer network (this is called node discovery). Then, the message needs to be properly routed handling obstacles like NAT traversal. All these things are handled by devp2p.
```

Comment:
**Node discovery** as a special thing that needs more elaboration

Reply:
Also NAT traversal not handly greatly afaik? So maybe refer here (or later) how libp2p can be used

Also (as todo maybe) worth mentioning there's some preliminary work on getting it to run on libp2p, murmur has clever workaround with replay of envelopes (IIRC)

Reply:
> Node discovery as a special thing that needs more elaboration

Right but I would link to devp2p spec for more details about that.

> Also NAT traversal not handly greatly afaik? So maybe refer here (or later) how libp2p can be used

Right. Again, I would not introduce things here which were not even considered when writing this protocol. We can maybe add a section that would shed some light on the fact that devp2p can be replaced with libp2p and Whisper with something else.

> Also (as todo maybe) worth mentioning there's some preliminary work on getting it to run on libp2p, murmur has clever workaround with replay of envelopes (IIRC)

Not sure it's the proper place. I would avoid any references to the implementations especially to POCs.